ui/src/app/course/components/editor/editor.component.ts

```diff
@@ -38,40 +66,126 @@ export class CourseEditorComponent implements OnInit, OnDestroy {
+			],
+			[formFields.date]: ['13/05/2019', [
+				Validators.pattern(COURSES_MODULE_DATE_REGEXP),
+				Validators.maxLength(10),
+				Validators.required,
+			]],
+			[formFields.duration]: [null]
+		});
+	}
+
+	private _loadCourse() {
+		this.subs.push(
+			this._courseService.getCourse(this.courseId)
+				.subscribe((course: TCourse) => {
+					this._store.dispatch(setCourseDetail({ payload: course }));
+					this._stopLoading();
+				}, this._handleNotFound.bind(this))
+		);
+	}
+
 	public onSubmit() {
 		if (confirm('U`re sure?')) {
-			this._onSubmit();
+		this._onSubmit();
 		}
 	}
 
 	private _onSubmit() {
-		const course = this.course.toJsonObject();
+		const course = this._getCourse();
+
 		if (this._isEditMode()) {
 			this._startLoading();
-			this.subs.push(this._courseService.updateCourse(this.courseId, course)
-				.subscribe((course: TCourse) => this._onSubscribeUpdate(course), this._handleNotFound.bind(this))
+			this.subs.push(
+				this._courseService.updateCourse(this.courseId, course)
+					.subscribe((course: TCourse) => this._onSubscribeUpdate(course), this._handleNotFound.bind(this))
 			);
 		} else {
-			this.subs.push(this._courseService.addCourse(course)
-				.subscribe((course: TCourse) => this._onSubscribeAdd(course), this._handleNotFound.bind(this))
+			this.subs.push(
+				this._courseService.addCourse(course)
+					.subscribe((course: TCourse) => this._onSubscribeAdd(course), this._handleNotFound.bind(this))
 			);
 		}
 	}
 
+	// region setters
+	set name(name: string) {
+		this.formGroup.get(formFields.name).setValue(name);
+	}
+
+	set duration(duration: number) {
+		this.formGroup.get(formFields.duration).setValue(duration);
+	}
+
+	set date(date: string) {
+		const dateTime = new Date(date || new Date().getFullYear() + '-01-01T00:00:00');
+		const prepareDate = this._datePipe.transform(dateTime, 'dd/MM/yyyy');
+		this.formGroup.get(formFields.date).setValue(prepareDate);
+	}
+
+	set description(description: string) {
+		this.formGroup.get(formFields.description).setValue(description);
+	}
+
+	// endregion
+
+	// region getters
+	get name(): string {
+		return this.formGroup.get(formFields.name).value;
+	}
+
+	get duration(): number {
+		return this.formGroup.get(formFields.duration).value;
+	}
+
+	get date(): string {
+		return this.formGroup.get(formFields.date).value;
+	}
+
+	get description(): string {
+		return this.formGroup.get(formFields.description).value;
+	}
+
+	// endregion
+
 	private _setCourse(course: TCourse) {
-		this.course = new CourseFormControl(course);
-		this._stopLoading();
+		this.name = course.name;
+		this.description = course.description;
+		this.date = course.date;
+		this.duration = course.length;
+		this.authors = course.authors;
+		this.isTopRated = course.isTopRated;
+
+		// console.log('course:', course);
+		// this.course = new CourseFormControl(course);
 		this._cdRef.markForCheck();
 	}
 
+	private _getCourse(): TCourse {
+		return {
+			id: this.courseId,
```

Comment:
And `this.formGroup.value` here

Reply:
done https://github.com/taraswww777/AngularGlobalMentoringProgram2019/commit/53f10d74